tests/testthat/test-checking.R

```diff
@@ -0,0 +1,33 @@
+test_that("check_dist_func works as expected", {
+  expect_true(check_dist_func(func = \(n) rgamma(n = n, shape = 2, scale = 2)))
+})
+
+test_that("check_dist_func errors when not a function", {
+  expect_error(
+    check_dist_func(func = "function"),
+    regexp = "(Assertion)*(func)*(failed)*(Must be a function, not)*(character)"
+  )
+})
+
+dist_func_error <- paste0(
+  "(generator must be a function with 1 argument(s) that returns)*",
+  "(non-negative numbers.)"
+)
+
+test_that("check_dist_func errors with non-negative non-numeric evaluation", {
+  expect_error(
+    check_dist_func(func = \(n) rep("i", n), dist_name = "generator"),
+    regexp = dist_func_error
```

Comment:
per above argument, feels over-prescriptive to me. If we want to guarantee that the error provided appropriate information specific to the argument (e.g. returned this type instead of numeric; returned a negative value), that's fine, but I think developers would want to avoid having to update these test values if they proposed wording tweaks to errors.